contracts/feature/Drop.sol

```diff
@@ -250,9 +243,14 @@ abstract contract Drop is IDrop, ExecutionContext {
         }
     }
 
-    /*///////////////////////////////////////////////////////////////
-        Virtual functions: to be implemented in derived contract
-    //////////////////////////////////////////////////////////////*/
+    /*////////////////////////////////////////////////////////////////////
+        Optional hooks that can be implemented in the derived contract
+    ///////////////////////////////////////////////////////////////////*/
+
+    /// @dev Exposes the ability to override the msg sender.
+    function _dropMsgSender() internal virtual returns (address) {
```

Comment:
only thing that's maybe a weird pattern is this. But saves us from inheriting ContextExecutable which adds burden for most ppl who don't use meta-tx.

let me know what you think @nkrishang @jakeloo

Reply:
yup this is similar to ERC721A pattern, maybe we can turn this into `_thirdwebMsgSender()` and we use it across the board or do you think this has to be isolated to each feature offering

Reply:
Pb is that it will clash if we declare it multiple times... So annoying. Hope solidity gets better overtime with this